microsoft/botframework-solutions
# Refactor calendar UT and fixed calendar bugs

## Description
1. Refactor the fake LUIS service to make it easier to fake an utterance.
2. Refactor the fake calendar service and user service. Now will create the fake data when test case initialize.
3. Fix the bug that meeting is created at past
4. Fix the bug that can't get next week meetings

## Related Issue
1. Meeting is created at past: https://fuselabs.visualstudio.com/Solutions/_queries/edit/26100/?triage=true
2. Can't get next week meetings: https://fuselabs.visualstudio.com/Solutions/_queries/edit/26102/?triage=true This is also a LUIS issue, the utterance is labelled as a wrong intent. Here will only fix the code issue that parse the "next week" to date failed.

## Testing Steps
1. Run all the test case
2. Ask the bot to "Create a meeting". Then when bot ask for date, input "Monday". The meeting should be created on next Monday.
3. Ask the bot "whats on my schedule next week" (need to add this string into FindCalendarEntry intent)
